apps/ledger-live-desktop/src/newArch/features/ModularDrawer/hooks/useAssetsData.ts

```diff
@@ -0,0 +1,34 @@
+import { useState, useCallback } from "react";
+import { AssetsDataWithPagination, useGetAssetsDataQuery } from "../data/state-manager/api";
+
+export interface UseAssetsDataResult {
+  data?: AssetsDataWithPagination;
+  isLoading: boolean;
+  error?: unknown;
+  hasMore: boolean;
+  cursor: string | undefined;
+  loadNext: () => void;
+}
+
+export function useAssetsData(): UseAssetsDataResult {
+  const [cursor, setCursor] = useState<string | undefined>(undefined);
+  const { data, isLoading, error } = useGetAssetsDataQuery({ cursor });
+
+  const loadNext = useCallback(() => {
```

Comment:
[ASK] do we plan to have loadPrevious?

Reply:
Not yet. We do not want to refresh the data while the drawer stays open when we scroll back up for example

Reply:
At some point, if we need it, we will be able to add it though